layer/main/asset.py

```diff
@@ -249,34 +244,14 @@ def _ui_progress_with_tracker(
     assert tracker
     asset_type = context.asset_type()
     assert asset_type
-    if asset_type == AssetType.MODEL:
-        if getting_asset_type == AssetType.DATASET:
-            tracker.mark_model_getting_dataset(
-                asset_name, getting_asset_name, from_cache
-            )
-        elif getting_asset_type == AssetType.MODEL:
-            tracker.mark_model_getting_model(
-                asset_name, getting_asset_name, state, from_cache
-            )
-    elif asset_type == AssetType.DATASET:
-        if getting_asset_type == AssetType.DATASET:
-            tracker.mark_dataset_getting_dataset(
-                asset_name, getting_asset_name, from_cache
-            )
-        elif getting_asset_type == AssetType.MODEL:
-            tracker.mark_dataset_getting_model(
-                asset_name, getting_asset_name, state, from_cache
-            )
+    tracker.mark_asset_downloading(
+        asset_type, asset_name, getting_asset_name, state, from_cache
+    )
     result = callback()
     if within_run:
-        if asset_type == AssetType.MODEL:
-            tracker.mark_model_training(asset_name)
-        elif asset_type == AssetType.DATASET:
-            tracker.mark_dataset_building(asset_name)
-    elif asset_type == AssetType.MODEL:
-        tracker.mark_model_loaded(asset_name)
-    elif asset_type == AssetType.DATASET:
-        tracker.mark_dataset_loaded(asset_name)
+        tracker.mark_running(asset_type, asset_name)
+    else:
+        tracker.mark_asset_downloaded(asset_type, asset_name)
```

Comment:
other one is `mark_uploading` so maybe either this one is `mark_downloaded` or other one is `mark_asset_uploading`?

Reply:
good point, done!